docs/UpdateButton.md

```diff
@@ -0,0 +1,354 @@
+| `onSuccess`      | Optional | `Function`  |        | A function to call when the mutation succeeds |
+| `onError`        | Optional | `Function`  |        | A function to call when the mutation fails    |
+
+`<UpdateButton>` also accepts the [Button props](./Buttons.md).
+
+## `data`
+
+The data used to update the record. Passed to the `dataProvider.update` method. This prop is required.
+
+{% raw %}
+```tsx
+import { TopToolbar, UpdateButton } from 'react-admin';
+
+const PostEditActions = () => (
+    <TopToolbar>
+        <UpdateButton label="Reset views" data={{ views: 0 }} />
+    </TopToolbar>
+);
+```
+{% endraw %}
+
+## `mutationMode`
+
+The mutation mode determines when the side effects (redirection, notifications, etc.) are executed:
+
+- `pessimistic`: The mutation is passed to the dataProvider first. When the dataProvider returns successfully, the mutation is applied locally, and the side effects are executed. 
+- `optimistic`: The mutation is applied locally and the side effects are executed immediately. Then the mutation is passed to the dataProvider. If the dataProvider returns successfully, nothing happens (as the mutation was already applied locally). If the dataProvider returns in error, the page is refreshed and an error notification is shown. 
+- `undoable` (default): The mutation is applied locally and the side effects are executed immediately. Then a notification is shown with an undo button. If the user clicks on undo, the mutation is never sent to the dataProvider, and the page is refreshed. Otherwise, after a 5 seconds delay, the mutation is passed to the dataProvider. If the dataProvider returns successfully, nothing happens (as the mutation was already applied locally). If the dataProvider returns in error, the page is refreshed and an error notification is shown.
+
+By default, the `<UpdateButton>` uses the `undoable` mutation mode. This is part of the "optimistic rendering" strategy of react-admin ; it makes user interactions more reactive. 
+
+You can change this default by setting the `mutationMode` prop. For instance, to remove the ability to undo the changes, use the `optimistic` mode:
+
+{% raw %}
+```jsx
+import { TopToolbar, UpdateButton } from 'react-admin';
+
+const PostEditActions = () => (
+    <TopToolbar>
+        <UpdateButton label="Reset views" data={{ views: 0 }} mutationMode="optimistic" />
+    </TopToolbar>
+);
+```
+{% endraw %}
+
+And to make the action blocking, and wait for the dataProvider response to continue, use the `pessimistic` mode:
+
+{% raw %}
+```jsx
+import { TopToolbar, UpdateButton } from 'react-admin';
+
+const PostEditActions = () => (
+    <TopToolbar>
+        <UpdateButton label="Reset views" data={{ views: 0 }} mutationMode="pessimistic" />
+    </TopToolbar>
+);
+```
+{% endraw %}
+
+
+**Tip**: When using any other mode than `undoable`, the `<UpdateButton>` displays a confirmation dialog before calling the dataProvider. 
```

Comment:
I wonder if we should provide a way to avoid having the confirmation dialog in pessimistic and optimistic modes too. Indeed, it makes sense for a Delete button, but not necessarily for an Update button... WDYT?